### PR TITLE
Remove backup trail

### LINKED ISSUE #11709: Upload of snapshot should check we did not reach the limit
### Bug description

Screenshot from 2022-07-27 20-21-29
Screenshot from 2022-07-27 20-22-12

### Steps to reproduce

Create multiple workspace snapshots

### Workspace affected

_No response_

### Expected behavior

_No response_

### Example repository

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by aledbf**
## Description

Remove the backup trail feature. The feature is present only on GCP. 
The reason for the removal is consistency with other providers and also to align it with PVC.

In traces, we see the error `Maximum number of snapshots (%d of %d) reached` meaning we are not going to upload a backup for the workspace and the next time it is open, it will contain the previous state, not the last one.

## Related Issue(s)

Fixes #11709

## Release Notes
```release-note
NONE
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

**Comment by roboquat** [comment]
@aledbf: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Review by MrSimonEmms — APPROVED** [review]
LGTM

It's worth rebasing from `main` as this may certainly fail due to #11288

**Review by Furisto — APPROVED** [review]

File: install/installer/pkg/config/v1/config.go [hunk 1]
```diff
@@ -218,13 +218,12 @@ type DatabaseCloudSQL struct {
 }
 
 type ObjectStorage struct {
-	InCluster          *bool                      `json:"inCluster,omitempty"`
-	S3                 *ObjectStorageS3           `json:"s3,omitempty"`
-	CloudStorage       *ObjectStorageCloudStorage `json:"cloudStorage,omitempty"`
-	Azure              *ObjectStorageAzure        `json:"azure,omitempty"`
-	MaximumBackupCount *int                       `json:"maximumBackupCount,omitempty"`
```
Comment (csweichel) [inline install/installer/pkg/config/v1/config.go:225]: This is a backwards incompatible change, afaik we'd break anyone who has a `maximumBackupCount` entry in their installer config.

We could leave the field and mark it as deprecated (a comment and validation). WDYT?

**Comment by csweichel** [comment]
Other than the backwards incompatible config change this LGTM.

File: install/installer/pkg/config/v1/config.go (hunk 1 above)
Comment (aledbf) [reply install/installer/pkg/config/v1/config.go:225]: Done

**Review by csweichel — APPROVED** [review]

**Pull request merged**